src/clang_parser.cpp

```diff
@@ -104,8 +104,19 @@ static SizedType get_sized_type(CXType clang_type)
       {
         return SizedType(Type::string, size);
       }
-      // TODO add support for arrays
-      return SizedType(Type::none, 0);
+
+      // Prevent unbounded recursion from get_sized_type call
+      if (elem_type.kind != CXType_ConstantArray)
+      {
+        auto type = get_sized_type(elem_type);
+        auto sized_type = SizedType(Type::array, size);
+        sized_type.is_pointer = true;
```

Comment:
I'm not sure either way about defining arrays as pointers at the moment - I need to think about it some more.

Reply:
This was probably a hack anyways, I'll look into if this is even needed.

Reply:
I've dropped this for now, I think I was doing this earlier to hook into some conditionals that checked for `is_pointer`, but i now do that by checking for `Type::array` so it seems it is no longer needed, and the runtime tests pass without this.